src/app/zap-templates/templates/app/cluster-objects.zapt

```diff
@@ -63,7 +69,55 @@ namespace {{name}} {
 } // namespace {{name}}
 {{/zcl_structs}}
 
-} // namespace {{asUpperCamelCase name}}
+namespace Commands {
+{{#zcl_commands}}
+namespace {{name}}CommandParams {
+constexpr CommandId kCommandId = {{asHex code 2}};
+
+enum FieldId {
+    {{#zcl_command_arguments}}
+    k{{asUpperCamelCase label}}FieldId = {{index}},
+    {{/zcl_command_arguments}}
+};
+
+struct Type {
+public:
+    {{#zcl_command_arguments}}
+    {{#if isArray}}
+    DataModel::List<{{#if_is_enum type}}{{asChipEnumType type}}{{else}}{{asChipZapType type}}{{/if_is_enum}}> {{asLowerCamelCase label}};
+    {{else}}
+    {{#if_is_enum type}}{{asChipEnumType type}}{{else}}{{asChipZapType type}}{{/if_is_enum}} {{asLowerCamelCase label}};
+    {{/if}}
+    {{/zcl_command_arguments}}
+
+    CHIP_ERROR Encode(TLV::TLVWriter &writer, uint64_t tag) const;
+    CHIP_ERROR Decode(TLV::TLVReader &reader);
+};
+
+{{#if command_contains_array}}
+struct DecodableType {
+public:
+    {{#zcl_command_arguments checkForDoubleNestedArray="true"}}
```

Comment:
```suggestion
    static constexpr chip::CommandId CommandId = {{asUpperCamelCase name}}::Id;

    {{#zcl_command_arguments checkForDoubleNestedArray="true"}}
```
Or so, to make #10216 work.  If that member name does not end up compiling because of the collision with the type name, we can use a different one, of course.